tests/e2e/csi_cns_telemetry.go

```diff
@@ -0,0 +1,207 @@
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc2.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, pvclaim2.Namespace, pvclaim2.Name, framework.Poll, time.Minute)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		persistentvolumes2, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvclaim2}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		volHandle2 := persistentvolumes2[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle2).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim2.Name, pvclaim2.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle2)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By(fmt.Sprintf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", volHandle2))
+		queryResult2, err := e2eVSphere.queryCNSVolumeWithResult(volHandle2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(queryResult2.Volumes) > 0).To(gomega.BeTrue())
+		framework.Logf("Cluster-distribution value on CNS is %s", queryResult2.Volumes[0].Metadata.ContainerCluster.ClusterDistribution)
+
+		queryResult3, err := e2eVSphere.queryCNSVolumeWithResult(volHandle1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(queryResult3.Volumes) > 0).To(gomega.BeTrue())
+		framework.Logf("Cluster-distribution value on CNS is %s", queryResult3.Volumes[0].Metadata.ContainerCluster.ClusterDistribution)
+
+		ginkgo.By("Verifying cluster-distribution value specified in secret is honored")
+		gomega.Expect(queryResult2.Volumes[0].Metadata.ContainerCluster.ClusterDistribution).Should(gomega.Equal(""), "Wrong/empty cluster-distribution name present on CNS")
+		gomega.Expect(queryResult3.Volumes[0].Metadata.ContainerCluster.ClusterDistribution).Should(gomega.Equal(""), "Wrong/empty cluster-distribution name present on CNS")
+
+		ginkgo.By("Setting the cluster-distribution value with escape character and special characters")
+		//Reset the cluster distribution value to original value
+		isSet = setClusterDistributionValue4Vanilla(ctx, client, "CSI-\tVanilla-#Test")
+		gomega.Expect(isSet).To(gomega.BeTrue())
+
+		defer func() {
+			//Reset the cluster distribution value to original value ("CSI-Vanilla")
+			isSet = setClusterDistributionValue4Vanilla(ctx, client, vanillaClusterDistribution)
+			gomega.Expect(isSet).To(gomega.BeTrue())
+		}()
+
+		ginkgo.By(fmt.Sprintf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", volHandle2))
+		queryResult2, err = e2eVSphere.queryCNSVolumeWithResult(volHandle2)
```

Comment:
Are you going to trigger full sync at this point? How do you expect the value to be updated immediately otherwise?

Reply:
The test environment is configured to run full sync every two min, and the tests wait for that duration and expect all the old PVCs to reflect the latest cluster-distribution value.

Corrected the timeout variable to reflect the fullSyncWaitTime.

Reply:
> The test environment is configured to run full sync every two min, and the tests wait for that duration and expect all the old PVCs to reflect the latest cluster-distribution value.

Can you point to the code where the test is waiting? From here it seems that we set cluster-distribution on line 182 and then immediately query CNS on line 186..